tasks/event.rb

```diff
@@ -0,0 +1,36 @@
+#!/opt/puppetlabs/puppet/bin/ruby
```

Comment:
This file is pretty big to not have any docs. Please add some docs to describe what it is doing and perhaps some inline ones as you see fit.

Reply:
The main docs are handled by Puppet Strings based on the JSON file, not much going on inside this file that's not described in JSON.